### ISSUE TITLE
infinite loop 

### ISSUE THREAD

**Issue by Lucuz91** [aws/aws-iot-device-sdk-python#24]
Hi SDK team,
I have a python application that runs inside a container. This application communicates with iot to exchange information. The following is the configuration inherent in the library.

```
self.myAWSIoTMQTTClient = AWSIoTMQTTClient (thron_key, useWebsocket = True)
self.myAWSIoTMQTTClient.configureEndpoint (host, 443)
self.myAWSIoTMQTTClient.configureCredentials (rootCAPath)

self.myAWSIoTMQTTClient.configureAutoReconnectBackoffTime (1, 128, 20)
self.myAWSIoTMQTTClient.configureOfflinePublishQueueing (-1) 
self.myAWSIoTMQTTClient.configureDrainingFrequency (2) 
self.myAWSIoTMQTTClient.configureConnectDisconnectTimeout (10) 
self.myAWSIoTMQTTClient.configureMQTTOperationTimeout (5) 

iot.myAWSIoTMQTTClient.subscribe ( "$ aws / things / thing_name / shadow / update / delta", 1, function_that_makes_things)

function_that_makes_things (client, userdata, message):
iot.myAWSIoTMQTTClient.publish ( "$ aws / things / thing_name / shadow / update", json.dumps (payload, separators = ( ',', ':')), 1)
```

The application works for hours without any problem, do not consume virtually no resources. But sometimes it freezes, saturate the CPU and unresponsive receives messages from iot. Interrupting the application I get this message:

```
Exception in thread Thread-1 (Most Likely raised During shutdown interpreter):
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
  File "/usr/lib/python2.7/threading.py", line 754, in run
  File "/usr/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 2345, in _thread_main
  File "/usr/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 1317, in loop_forever
  File "/usr/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 862, looped
  File "/usr/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 1124, in loop_read
  File "/usr/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 1464, in _packet_read
<Type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'error'
```

It seems that the problem is within the PAHO library. This problem seems to have been solved by a bugfix in version 1.1 [https://github.com/eclipse/paho.mqtt.python/blob/master/ChangeLog.txt]
The place seems to be the version 1.0, are no plans to upgrade to 1.2 or 1.1? Or am I using the library in the wrong way?

**Comment by chaurah** [comment]
Hi @Lucuz91,
     Thanks for pointing this out. We already have a plan to upgrade to the latest version of the Paho SDK in a future release. Unfortunately, there's no timeline on that release at the moment. I will update this ticket when the release is out.

Rahul

**Comment by Lucuz91** [comment]
Hello,
i think understand what creates the loop. the problem is a limitation of the WebSocket connection, that is 24 hours.
We confirmed that after exactly 24 hours after application start, CPU go at 100%.
We found this limitation in documentation IOT http://docs.aws.amazon.com/iot/latest/developerguide/iot-limits.html, but we would have expected a fall of the connection, or the connection restart,
not the loop of thread.

**Comment by liuszeng** [comment]
Hi @Lucuz91 ,

Thank you very much for providing the information. We will keep investigating into this and try to duplicate the issue on our side. In the meantime, have you seen any log output when you are experiencing the issue (WebSocket lifetime connection closing)?

Thanks,
Liusu

Thanks,
Liusu

**Comment by Lucuz91** [comment]
Hi @liuszeng ,

I don't see any log, only thing i see is the cpu at 100%

**Comment by hedeshianaws** [comment]
@Lucuz91 There have been some changes and bug fixes in v1.2.0 which may affect this issue. Are you still seeing the issue?

**Comment by hedeshianaws** [comment]
Closing due to inactivity. Please re-open the issue if you are able to reproduce the issue.
